src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -1468,7 +1470,8 @@ def _dropna(
             raise ValueError("invalid how option: %s" % how)
         if subset is not None:
             if axis == 1:
-                indices = self.index.get_indexer_for(subset)
+                # TODO: SNOW-1372242: Remove instances of to_pandas when lazy index is implemented
+                indices = self.index.to_pandas().get_indexer_for(subset)
```

Comment:
Add `get_indexer_for` method to `Index` object and do the `to_pandas` there.
```suggestion
                indices = self.index.get_indexer_for(subset)
```